src/rpc/util.cpp

```diff
@@ -677,6 +666,44 @@ UniValue RPCHelpMan::GetArgMap() const
     return arr;
 }
 
+void RPCArg::MatchesType(const UniValue& request) const
+{
+    if (m_opts.skip_type_check) return;
+    if (IsOptional() && request.isNull()) return;
+    switch (m_type) {
+    case Type::STR_HEX:
+    case Type::STR: {
+        RPCTypeCheckArgument(request, UniValue::VSTR);
```

Comment:
Small cleanup idea:
what if we remove `RPCTypeCheckArgument` and just call `request.checkType(UniValue::VSTR)`?
(would just need to make `checkType` public)

Reply:
`checkType` is used for json type checks. As mentioned previously you wouldn't be able to check for other types, such as hex strings (https://github.com/bitcoin/bitcoin/pull/26039#discussion_r971058812) in the future. So I'd like to leave them copied for now. If you feel strongly, it might be better to do the replacement in `RPCTypeCheckArgument`, which can be done separate form this pull request?

Reply:
ha, I completely forgot that I mentioned it before. Sorry for that. nvm then.